pkg/controller/scheduler.go

```diff
@@ -247,7 +251,72 @@ func (c *Controller) advanceCanary(name string, namespace string, skipLivenessCh
 		}
 	}
 
-	// increase canary traffic percentage
+	// canary fix routing: A/B testing
+	if len(cd.Spec.CanaryAnalysis.Match) > 0 {
+		// route traffic to canary and increment iterations
+		if cd.Spec.CanaryAnalysis.Iterations > cd.Status.Iterations {
+			if err := meshRouter.SetRoutes(cd, 0, 100); err != nil {
+				c.recordEventWarningf(cd, "%v", err)
+				return
+			}
+			c.recorder.SetWeight(cd, 100, 0)
```

Comment:
:sweat_smile: : shouldn't recorder do a `SetWeight(cd, 0,100) ` to reflect the meshRouter.SetRoutes ?

Reply:
good catch!